cmd/zt_sync_upload_windows_test.go

```diff
@@ -88,3 +87,35 @@ func (s *cmdIntegrationSuite) TestSyncUploadWithIncludeAttrFlag(c *chk.C) {
 		validateUploadTransfersAreScheduled(c, "", "", filesToInclude, mockedRPC)
 	})
 }
+
+// Asserting that name filter and attribute filter are ANDed
+func (s *cmdIntegrationSuite) TestSyncUploadWithIncludeAndIncludeAttrFlags(c *chk.C) {
+	bsu := getBSU()
+
+	srcDirName := scenarioHelper{}.generateLocalDirectory(c)
+	scenarioHelper{}.generateCommonRemoteScenarioForLocal(c, srcDirName, "")
+
+	filesToInclude := []string{"file1.txt", "file2.txt", "file3.pdf"}
+	scenarioHelper{}.generateLocalFilesFromList(c, srcDirName, filesToInclude)
+	includeString := "*.pdf;file*.txt"
+	includeAttrsStr := "H;I;S"
+	attrList := []string{"H", "I", "C"}
+	scenarioHelper{}.setAttributesForLocalFiles(c, srcDirName, filesToInclude, attrList)
```

Comment:
I'm not sure if this test is actually making any assertion that would distinguish between ANDing and ORing - since all filesToInclude match both the include string and the attrs string.  It would be nice to have at least 3 files in the file set, with the following characteristics: 
* One that only matches includeString, but not the attributes
* One that only matches an attribute, but not the include string
* And one that matches both (and the test would assert that only this one gets transferred)

To setup, you've had to maybe change the include string to be just *.txt, and then pass filesToInclude[1:] to setAttributesForLocalFiles.  The expected transfer would be just filesToInclude[1:2], since only that matches both.

(And I guess "filesToInclude" is not really the right variable name anymore ;-)

Reply:
Ditto, this only creates one set of files with both sets of includable items.

---
In reply to: [304708701](https://github.com/Azure/azure-storage-azcopy/pull/501#discussion_r304708701) [](ancestors = 304708701)